rust/lance-index/src/vector/hnsw/builder.rs

```diff
@@ -159,7 +159,7 @@ impl HNSWBuilder {
         //  }
         // ```
         for cur_level in self.levels.iter().rev().take(levels_to_search) {
-            let candidates = beam_search(cur_level, &ep, vector, self.ef_construction)?;
+            let candidates = beam_search(cur_level, &ep, vector, 1)?;
             ep = if self.use_select_heuristic {
                 select_neighbors_heuristic(cur_level, query, &candidates, 1, self.extend_candidates)
                     .map(|(_, id)| id)
```

Comment:
```suggestion
```
these could be all removed as the `beam_search` returns only one candidate

Reply:
Good point, Actually, is `select_heuristic(exstend_candidate=False)` just `select_simple`?

Reply:
Fixed.

Reply:
> Good point, Actually, is `select_heuristic(exstend_candidate=False)` just `select_simple`?

for the current impl it's the same but likely it doesn't follow the paper